### PR TITLE
Make order=None work like default order

### LINKED ISSUE #590: cupy.array(cupy_array, oder=None) raises error
When I do this:
```python
>>> x = cupy.ones(3)
>>> xx = cupy.array(x, order=None)
```
I get this traceback:
```

  File "[...]/cupy/cupy/creation/from_data.py", line 41, in array
    return core.array(obj, dtype, copy, order, subok, ndmin)

  File "cupy/core/core.pyx", line 2026, in cupy.core.core.array
    cpdef ndarray array(obj, dtype=None, bint copy=True, str order='K',

  File "cupy/core/core.pyx", line 2039, in cupy.core.core.array
    a = src.astype(dtype, order=order, copy=copy)

  File "cupy/core/core.pyx", line 276, in cupy.core.core.ndarray.astype
    cpdef ndarray astype(

  File "cupy/core/core.pyx", line 313, in cupy.core.core.ndarray.astype
    raise TypeError('order not understood')

TypeError: order not understood
```

### PR DESCRIPTION
**Pull request by kohr-h**
Fixes #590

### CODE REVIEW

File: cupy/core/core.pyx [hunk 1]
```diff
@@ -309,13 +309,15 @@ cdef class ndarray:
...
+        if order is None:
+            order = 'K'
         if order not in ['C', 'F', 'A', 'K']:
             raise TypeError('order not understood')
 
         dtype = numpy.dtype(dtype)
         if dtype == self.dtype:
             if not copy and (
-                    order == 'K' or
+                    order in 'K' or
```
Comment (niboshi) [inline cupy/core/core.pyx]: Please use `==` operator.

File: tests/cupy_tests/core_tests/test_ndarray.py [hunk 2]
```diff
@@ -63,6 +63,13 @@ def test_order(self):
         self.assertTrue(a.flags.f_contiguous)
         self.assertTrue(not a.flags.c_contiguous)
 
+    def test_order_None(self):
```
Comment (niboshi) [inline tests/cupy_tests/core_tests/test_ndarray.py]: Please use lower case `none` for a method name.

File: tests/cupy_tests/core_tests/test_ndarray.py [hunk 3]
```diff
@@ -63,6 +63,13 @@ def test_order(self):
...
+        a = core.ndarray(self.shape, order=None)
+        a_cpu = numpy.ndarray(self.shape, order=None)
+        self.assertTupleEqual(a.strides, a_cpu.strides)
+        self.assertTrue(a.flags.c_contiguous)
+        self.assertTrue(not a.flags.f_contiguous)
```
Comment (niboshi) [inline tests/cupy_tests/core_tests/test_ndarray.py]: Could you add assertions to test `a_cpu`'s `c_contiguous` and `f_contiguous` as well? That way we can be notified when NumPy changes the behavior.

Comment (kohr-h) [reply tests/cupy_tests/core_tests/test_ndarray.py]: Are you okay with switching to test `a.flags.c_contiguous == a_cpu.flags.c_contiguous` and similarly for `'F'`? I guess the purpose is to track Numpy behavior.

File: tests/cupy_tests/core_tests/test_ndarray.py (hunk 2 above)
Comment (kohr-h) [reply tests/cupy_tests/core_tests/test_ndarray.py]: OK

**Comment by kohr-h** [comment]
I changed the code according to the comments.

File: tests/cupy_tests/core_tests/test_ndarray.py (hunk 3 above)
Comment (niboshi) [reply tests/cupy_tests/core_tests/test_ndarray.py]: Yes, that's also good.

**Comment by niboshi** [comment]
Thank you for fix!
jenkins, test this please

**Comment by niboshi** [comment]
LGTM.
Thank you very much!

**Pull request merged**